File: garnet/lib/perfmon/test_main.cc
// Copyright 2019 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <gtest/gtest.h>
#include <src/lib/fxl/command_line.h>
#include <src/lib/fxl/log_settings.h>
#include <src/lib/fxl/log_settings_command_line.h>

// Provide our own main so that --verbose,etc. are recognized.
int main(int argc, char** argv) {
  auto cl = fxl::CommandLineFromArgcArgv(argc, argv);
  if (!fxl::SetLogSettingsFromCommandLine(cl))
    return EXIT_FAILURE;

  testing::InitGoogleTest(&argc, argv);

  return RUN_ALL_TESTS();
}

File: src/experiences/session_shells/ermine/next/lib/test_main.dart
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:flutter_driver/driver_extension.dart';

import 'package:next/src/services/shortcuts_service.dart';

import 'main.dart' as entrypoint;

Future<void> main() async {
  enableFlutterDriverExtension(
      enableTextEntryEmulation: false,
      handler: (String? data) async {
        if (data != null && ShortcutsService.flutterDriverHandler != null) {
          return ShortcutsService.flutterDriverHandler!.call(data);
        }
        return '';
      });
  return entrypoint.main();
}
